crates/autopilot/src/run_loop.rs

```diff
@@ -621,28 +756,36 @@ impl RunLoop {
     /// transaction has been mined.
     async fn settle(
         &self,
-        driver: &infra::Driver,
-        solved: &competition::SolutionWithId,
+        driver: Arc<infra::Driver>,
```

Comment:
Can stay as it was.
```suggestion
        driver: &infra::Driver,
```

Reply:
This function is called inside the `async move`, so I believe this is not possible.

https://github.com/cowprotocol/services/blob/bb39916308a2204075f49cf92c4aa203508cab1d/crates/autopilot/src/run_loop.rs#L273-L279

Reply:
I tested it out locally and it compiled. The reason is that you move an owned `Arc<T>` into the future so the reference is guaranteed to outlive the future since the `Arc<T>` outlives the function call.